x-pack/solutions/security/plugins/security_solution/public/common/components/query_bar/index.tsx

```diff
@@ -160,11 +160,11 @@ export const QueryBar = memo<QueryBarComponentProps>(
         createDataView();
       }
       return () => {
-        if (dv?.id && !newDataViewPickerEnabled) {
```

Comment:
we can't disable clearing cache here via feature flag, because rule creation also uses this component and need to create ad hoc data views

Reply:
Can you actually add a comment in line saying this?
`// Cache needs to be cleared in certain instances, like rule creation, for ad hoc dataviews`...or something like that